### DISCUSSION TITLE
jotai not compiling with Expo web

### DISCUSSION THREAD

**Discussion by bidah** [pmndrs/jotai#1717] [Q&A]
when compiling jotai for Expo web `npx expo export:web` with the config to transpile jotai on webpack config as follows

```
const createExpoWebpackConfigAsync = require("@expo/webpack-config");

module.exports = async function (env, argv) {
  const config = await createExpoWebpackConfigAsync(
    {
      ...env,
      babel: {
        dangerouslyAddModulePathsToTranspile: ["nativewind", "jotai"],
      },
    },
    argv
  );

  config.module.rules.push({
    test: /\.css$/i,
    use: ["postcss-loader"],
  });

  return config;
};

```

still getting the compile error
```
index.js:1 
        
       ./node_modules/jotai/esm/index.mjs 71:13
Module parse failed: Unexpected token (71:13)
File was processed with these loaders:
 * ./node_modules/babel-loader/lib/index.js
You may need an additional loader to handle the result of these loaders.
|   var stateListeners;
|   var mountedAtoms;
>   if ((import.meta.env && import.meta.env.MODE) !== "production") {
|     stateListeners = new Set();
|     mountedAtoms = new Set();
```

Any ideas to fix this are welcomed.

**Comment by dai-shi** [comment]
Oh, so they pick "import" condition, but they don't understand `import.meta`. That's unexpected. 

We should let them pick CJS or "module". Does anyone know how to do it?
(If "module" is useful we shouldn't drop it. @barelyhuman )

**Reply by barelyhuman** [reply]
@bidah Since no versions are mentioned, I'm going to assume this behaviour is from webpack 4 and if it is, you can try adding this to the config. 

If it's picking up `.mjs` it's a loader mismatch and you'll have to add that yourself.

```
config.module.rules.push({
    test: /\.mjs$/,
    include: /node_modules/,
    type: "javascript/auto"
  });

```

If this doesn't work, please do mention the expo version so I can dig deeper

**Reply by bidah** [reply]
Hey @barelyhuman thanks helping out.
Added the config you mentioned to webpack but still getting the same error.
Im on expo v `~47.0.9` and babel-core `^7.12.9`

**Reply by barelyhuman** [reply]
So, for me , I could get it to export and work without having to add `jotai` to `dangerouslyAddModulePathsToTranspile` , did you try to remove it and try a compile? Is that also giving an error ? because it seems to be working for me. 

Do try that if you haven't, if not could you help with a reproduction repo as well?

**Reply by bidah** [reply]
happening too without it. will make the reproduction repo and share here today

**Reply by bidah** [reply]
so making the reproduction example found the error. It was working, so the issue was that on my code I did an auto import and the import was as follows `import { useAtom } from "jotai/esm"`. That was breaking the build.

**Reply by barelyhuman** [reply]
Oh okay, glad to know it worked out
